File: blog/src/main/java/com/xmq/entity/Article.java
package com.xmq.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.util.Date;
import lombok.Data;

/**
 * @TableName article
 */
@TableName(value ="article")
@Data
public class Article implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String content;

    private String summary;

    private String coverImage;

    private Integer viewCount;

    private Integer commentCount;

    private Integer likeCount;

    private Integer collectionCount;

    private Integer status;

    private Long authorId;

    private Integer isTop;

    private Integer isBroadcast;

    private Date createdTime;

    private Date updatedTime;

    private Integer isDeleted;

    private static final long serialVersionUID = 1L;
}

File: vue/user/src/types/article.ts
export interface Article {
  id: number
  title: string
  content: string
  summary: string
  coverImage: string
  viewCount: number
  commentCount: number
  likeCount: number
  collectionCount: number
  status: number
  authorId: number
  createdTime: string
  updatedTime: string
  isDeleted: number
}

export interface ArticleHotVO {
  id: number
  title: string
  summary: string
  coverImage: string
  viewCount: number
  likeCount: number
  commentCount: number
  collectionCount: number
  periodCount: number
  authorId: number
  authorName: string
  createdTime: string
}

export interface ArticleQueryDTO {
  page?: number
  size?: number
  keyword?: string
  status?: number
}

export interface HotArticleParams {
  type: 'views' | 'likes'  // 修改为只有 views 和 likes
  period: string
  limit?: number
}

export interface Comment {
  id: number
  articleId: number
  content: string
  userId: number
  username: string
  userAvatar?: string
  parentId?: number
  replyTo?: string
  createdTime: string
  replies?: Comment[]
}

export interface ApiResponse<T> {
  code: number
  message: string
  data: T
} 